test/Algebra/Graph/Test/Label.hs

```diff
@@ -0,0 +1,144 @@
+{-# LANGUAGE OverloadedLists #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Test.Label
+-- License    : MIT (see the file LICENSE)
+-- Stability  : experimental
+--
+-- Testsuite for "Algebra.Graph.Label".
+-----------------------------------------------------------------------------
+{-
+module Algebra.Graph.Test.Label (
+  -- * Testsuite
+  testLabel
+  ) where
+-}
+module Algebra.Graph.Test.Label where
+
+import Algebra.Graph.Test hiding (NonNegative)
+import Algebra.Graph.Label
+import Data.Monoid
+
+type UnaryFn a = a -> a
+type BinaryFn a = a -> a -> a
+type Plus a = BinaryFn a
+type Mult a = BinaryFn a
+type Star a = UnaryFn a
+type Identity a = a
+type Zero a = a
+type One a = a
+type Annhilator a = a
+
+associative :: Eq a => BinaryFn a -> a -> a -> a -> Property
+associative f a b c = (a `f` b) `f` c == a `f` (b `f` c) // "Associative property"
+
+commutative :: Eq a => BinaryFn a -> a -> a -> Property
+commutative f a b = a `f` b == b `f` a // "Commutative property"
+
+idempotance :: Eq a => BinaryFn a -> a -> Property
+idempotance f a = a `f` a == a // "Idempotence property"
+
+annhilator :: Eq a => BinaryFn a -> Annhilator a -> a -> Property
+annhilator f z a = conjoin
+  [ a `f` z == z // "Annhilator property 1"
+  , z `f` a == z // "Annhilator property 2" ]
```

Comment:
These properties are usually referred to as "left" (`0 * x = 0`) and "right" (`x * 0 = 0`).

Reply:
Same for some others below.